jme3-core/src/main/java/com/jme3/anim/tween/action/BlendAction.java

```diff
@@ -1,129 +1,265 @@
+/*
+ * Copyright (c) 2009-2021 jMonkeyEngine
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ * * Redistributions of source code must retain the above copyright
+ *   notice, this list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright
+ *   notice, this list of conditions and the following disclaimer in the
+ *   documentation and/or other materials provided with the distribution.
+ *
+ * * Neither the name of 'jMonkeyEngine' nor the names of its contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+ * TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
 package com.jme3.anim.tween.action;
 
 import com.jme3.anim.util.HasLocalTransform;
 import com.jme3.math.Transform;
-
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.Map;
 
+/**
+ * A concrete implementation used to blend between 2 actions #{@link BlendAction#firstActiveIndex}, #{@link BlendAction#secondActiveIndex} & apply them on their targets,
+ * blending between 2 actions is done by interpolating the set of transforms serially starting from the firstAction transforms, moving on to the second one transforms,
+ * using the blendWeight as a delta value, then interpolating the resulted value with the current target local transforms,
```

Comment:
@stephengold I would like to discuss some of these changes here, before shipping the documentation to a newly separated PR.

I looked up the original diagram created by Nehon on forums, the most righteous expression that could be used to describe the "blend weight" is "blending driver value", What do you think?

Reply:
I think I have now remembered why I chose to use "delta" to describe the blendweight, the `ClipAction` class calls the `Transform#interpolateTransforms(Transform t1, Transform t2, float delta)` which passes the weight as the delta value of interpolation!!

I will let you decide on this, but in my opinion, it's mathematically a delta value.

Reply:
I still think "delta" is confusing, both here and in "Transform.java". It's the parameter of the interpolation, which controls the relative weight given to each input, right? To me "delta" describes the difference between 2 inputs.

Reply:
> I looked up the original diagram created by Nehon on forums, the most righteous expression that could be used to describe the "blend weight" is "blending driver value", What do you think?

It's good to see the diagram and understand what a blending driver is, but in reality, people won't look-up this diagram, it must be crystal clear "What's the blending?" and "What's the weight of blending?" which could summarised in this sentence "a delta value (or scale factor) for animation transformation data interpolation".

Reply:
> I still think "delta" is confusing, both here and in "Transform.java". It's the parameter of the interpolation, which controls the relative weight given to each input, right? To me "delta" describes the difference between 2 inputs.

Okay, so, does the "scale factor of interpolation" satisfies a good description for the blending weight?

Reply:
Btw, the parameter of linear interpolation is calculated mathematically by dividing the value of "x(in) - x0" by the value of "x1 - x0", so that is why it is called delta, IMHO, it describes the change of x(in) with respect to the origin point (x0) in the way to x1.

Reply:
It's just the "value for interpolation".  I agree delta makes no sense.  0 means the first value, 0.5 means halfway between the two values, 1 means the second value.

re: "it describes the change of x(in) with respect to the origin point (x0) in the way to x1." ...which would be fine if it wasn't rescaled to be between x0 and x1 again.  It's not a delta.

JME tends to call it "scale" but that's not really correct either.

Reply:
Ok, I will use "value for interpolation" instead of "delta" to avoid confusion. Thanks.